sdk/nodejs/runtime/invoke.ts

```diff
@@ -49,10 +85,11 @@ export async function invoke(tok: string, props: Inputs, opts?: InvokeOptions):
         const monitor: any = getMonitor();
 
         let providerRef: string | undefined;
-        if (opts.provider !== undefined) {
-            const providerURN = await opts.provider.urn.promise();
-            const providerID = await opts.provider.id.promise() || unknownValue;
-            providerRef = `${providerURN}::${providerID}`;
+        if (ProviderRef.isInstance(provider)) {
+            providerRef = provider.getValue();
+        }
+        else if (Resource.isInstance(provider)) {
+            providerRef = (await ProviderRef.get(provider)).getValue();
         }
```

Comment:
We're in the `async` invoke impl.  So it's totally fine to get the ref for a provider asynchronously here.